include/RAJA/util/SoAPtr.hpp

```diff
@@ -38,18 +41,37 @@ namespace detail
  */
 template <typename T,
           typename mempool = RAJA::basic_mempool::MemPool<
-              RAJA::basic_mempool::generic_allocator> >
+              RAJA::basic_mempool::generic_allocator>,
+          typename accessor = DefaultAccessor >
 class SoAPtr
 {
-  using value_type = T;
+  template < typename, typename, typename >
+  friend class SoAPtr; // fiend other instantiations of this class
```

Comment:
Simple typo, but out of curiosity why are we doing this kind of self-friending?

```suggestion
  friend class SoAPtr; // friend other instantiations of this class
```

Reply:
Different template instantiations of the a class template are not the same type.